docs/whats-new/preview-4.md

```diff
@@ -0,0 +1,647 @@
+2. The port that serves the OTLP grpc endpoint (18889)
+
+That will bring up a dashboard that you can use view logs, metrics, and traces from your applications. Here's a [sample application](https://learn.microsoft.com/en-us/samples/dotnet/aspire-samples/aspire-standalone-dashboard/) that sends telemetry data to the dashboard.
+
+### Dashboard shortcuts
+
+The Aspire Dashboard now supports keyboard navigation via keyboard shortcuts. Click <kbd>Shift</kbd>+<kbd>?</kbd> to display the list of available shortcuts:
+
+:::image type="content" source="media/preview-4/dashboard-shortcuts.png" lightbox="media/preview-4/dashboard-shortcuts.png" alt-text=".NET Aspire Dashboard shortcuts dialog.":::
+
+### Metrics table view
+
+With preview 4, we have introduced a screen reader compatible table view for display of metrics data. This has similar options, as the graph, for filters and selecting the duration of time range for metrics display. The default data display is set to "Only show value updates" and can be toggled to display all data points.
+
+![image](https://github.com/dotnet/docs-aspire/assets/102772054/4622cf3a-dcf2-4149-9636-d1bef8184c5c)
+
+## Databases, EntityFramework and Aspire
+
+### New Enrich methods
+
+Preview 4 introduces new methods for configuring Entity Framework. The existing `Add[Provider]DbContext()` methods used to register and configure `DbContext` classes are not sufficient for advanced cases like using a different lifetime scope, using custom service types, or configuring the underlying data sources.
+
+To solve these advanced scenarios, new methods named `Enrich[Provider]DbContext()` have been added. These methods do not register the `DbContext` and expect you to do so before invoking them.
+
+Usage example:
+
+```csharp
+var connectionString = builder.Configuration.GetConnectionString("catalogdb");
+builder.Services.AddDbContextPool<CatalogDbContext>(dbContextOptionsBuilder => dbContextOptionsBuilder.UseSqlServer(connectionString));
+builder.EnrichSqlServerDbContext<CatalogDbContext>();
+```
+
+These methods will still configure command retries, health checks, logging and telemetry.
+
+### Changes to previously existing methods
+
+Since these new methods provide a simpler way to configure the `DbContext`, the already existing ones (`Add[Provider]DbContext()`) have been simplified.
+
+1- They don't provide a way to disable connection pooling through settings anymore. Instead register the `DbContext` with connection pooling disabled and call the corresponding `Enrich` method.
+2- The `int MaxRetryCount` settings were removed and replaced by a `bool Retry` flag that and uses the default settings. It is enabled by default. To change the command retries count to a custom value please configure the `DbContext` registration using the specific provider options and invoke the corresponding `Enrich` method.  
+
+### Entity Framework Migrations
+
+TODO: Migration tooling guidance (@JamesNK)
```

Comment:
@JamesNK I need this from you this week. I'm assuming we want to move [this sample](https://github.com/dotnet/aspire/tree/main/playground/DatabaseMigration) to the official samples repo and write a small blurb about adding migrations (which has improved).

Reply:
Sample PR: https://github.com/dotnet/aspire-samples/pull/142

Reply:
Excellent! Just need the blurb about what the guidance is based on what we discussed with the EF team (and point to future improvements on EF issues?)